cmd/cloud-controller-manager/app/core.go

```diff
@@ -98,12 +101,23 @@ func startRouteController(ctx *cloudcontrollerconfig.CompletedConfig, cloud clou
 		klog.Warning("configure-cloud-routes is set, but cloud provider does not support routes. Will not configure cloud provider routes.")
 		return nil, false, nil
 	}
-	var clusterCIDR *net.IPNet
-	var err error
+	clusterCIDRs := make([]*net.IPNet, 0)
 	if len(strings.TrimSpace(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)) != 0 {
-		_, clusterCIDR, err = net.ParseCIDR(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)
-		if err != nil {
-			klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+		clusterCIDRList := strings.Split(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, ",")
```

Comment:
This seems like it should be a common helper function with node IPAM parseing and validation

Reply:
Could we put this into a new pkg/util/network/network.go and use it from here and  cmd/kube-controller-manager/app/core.go?

Reply:
Ideally we use the following IPNetSlice feature instead: https://github.com/spf13/pflag/pull/170
I've pinged the spf13 folks to get that merged. 
This is also mentioned in the KEP https://github.com/kubernetes/enhancements/pull/808/files#diff-df06b6d8f252f5f3711baaaf155ed3eaR611

Reply:
I agree on `IPNetSlice`, will look into it.. But for now i will go @dcbw proposal..